server/rooms.js

```diff
@@ -1610,55 +1610,86 @@ let Rooms = Object.assign(getRoom, {
+		if (gameType !== 'multi' && gameType !== 'free-for-all') {
+			if (players[2] || players[3]) {
+				throw new Error(`Four players were provided, but the format is a 2 player format.`);
+			} else {
+				delete players[2];
+				delete players[3];
+			}
+		}
+		for (const [i, p1] of players.entries()) {
+			if (!p1) continue;
+			for (const p2 of players.slice(i + 1)) {
+				if (p1 === p2) throw new Error(`Players can't battle themselves`);
+			}
+		}
+
+		for (const p of players) {
+			if (p) Ladders.cancelSearches(p);
+		}
 
 		if (Rooms.global.lockdown === true) {
-			if (p1) p1.popup("The server is restarting. Battles will be available again in a few minutes.");
-			if (p2) p2.popup("The server is restarting. Battles will be available again in a few minutes.");
+			for (const p of players) {
+				if (p) p.popup("The server is restarting. Battles will be available again in a few minutes.");
+			}
 			return;
 		}
 
-		// @ts-ignore
-		if (p1 && p2 && (p1.specialNextBattle || p2.specialNextBattle)) {
+		if (!players.includes(null)) {
 			// @ts-ignore
-			const p1special = p1.specialNextBattle, p2special = p2.specialNextBattle;
-			// @ts-ignore
-			p1.specialNextBattle = null;
-			// @ts-ignore
-			p2.specialNextBattle = null;
-
-			if (p1special !== p2special) {
-				p1.popup(`Your special battle settings don't match: "${p1special}" and "${p2special}"`);
-				p2.popup(`Your special battle settings don't match: "${p1special}" and "${p2special}"`);
+			const p1Special = players[0].specialNextBattle;
+			let mismatch = `"${p1Special}"`;
+			for (const p of players) {
+				// @ts-ignore
+				if (p && p.specialNextBattle !== p1Special) {
+					// @ts-ignore
+					mismatch += ` vs. "${p.specialNextBattle}"`;
+					break;
+				}
+			}
+			if (mismatch !== `"${p1Special}"`) {
+				for (const p of players) {
+					// @ts-ignore p is not null
+					p.popup(`Your special battle settings don't match: ${mismatch}`);
+				}
 				return;
 			}
-			options.ratedMessage = p1special;
+			options.ratedMessage = p1Special;
 		}
 
 		const roomid = Rooms.global.prepBattleRoom(formatid);
 		options.format = formatid;
-		// options.rated is a number representing the lower player rating, for searching purposes
+		// options.rated is a number representing the lowest player rating, for searching purposes
 		// options.rated < 0 or falsy means "unrated", and will be converted to 0 here
 		// options.rated === true is converted to 1 (used in tests sometimes)
 		options.rated = Math.max(+options.rated || 0, 0);
+		const p1 = players[0];
+		const p2 = players[1];
 		const p1name = p1 ? p1.name : "Player 1";
 		const p2name = p2 ? p2.name : "Player 2";
-		const room = Rooms.createGameRoom(roomid, "" + p1name + " vs. " + p2name, options);
+		let roomTitle;
+		if (gameType === 'multi') {
+			roomTitle = `Team ${p1name} vs. Team ${p2name}`;
+		} else if (gameType === 'free-for-all') {
+			// p1 vs. p2 vs. p3 vs. p4 is too long of a title
+			roomTitle = "An all-out brawl!";
```

Comment:
```suggestion
			roomTitle = "Free-for-all";
```

I'd prefer not to editorialize that much in a title.

Reply:
That's redundant with the roomid that goes right above the title, "battle-gen7freeforall-#", which is why I chose not to go with that in the first place.

Reply:
Can we find something in-between, or at least something other than "Free-for-all"?

Reply:
Room IDs don't go above the title... the title replaces the room ID.

Reply:
Like for instance, in the Dev room, `development` is the room ID, and `Development` is the title.

Reply:
`${p1name} etc`?

Reply:
http://prntscr.com/myqsxj The title goes on the bottom.

Reply:
Fair.

`` `${p1name} etc` ``?

Reply:
`` `${p1name} and friends` ``?